crates/rspack_ids/src/id_helpers.rs

```diff
@@ -253,7 +253,7 @@ pub fn assign_deterministic_ids<T: Copy>(
   mut items: Vec<T>,
   get_name: impl Fn(T) -> String,
   comparator: impl Fn(&T, &T) -> Ordering,
-  mut assign_id: impl FnMut(T, String) -> bool,
+  mut assign_id: impl FnMut(T, usize) -> bool,
```

Comment:
I converted `usize` to `String` at the usage, so the rest type should remain unchanged.  this alignment is required by `mangleExportsPlugin`.